dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/RelationshipRestRepository.java

```diff
@@ -60,4 +91,85 @@ public RelationshipRest findOne(Context context, Integer integer) {
     public DSpaceResource<RelationshipRest> wrapResource(RelationshipRest model, String... rels) {
         return new RelationshipResource(model, utils, rels);
     }
+
+    protected RelationshipRest createAndReturn(Context context, List<DSpaceObject> list)
+        throws AuthorizeException, SQLException, RepositoryMethodNotImplementedException {
+
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+
+        Relationship relationship = new Relationship();
+        if (list.size() == 2 && list.get(0).getType() == Constants.ITEM && list.get(1).getType() == Constants.ITEM) {
+            Item leftItem = (Item) list.get(0);
+            Item rightItem = (Item) list.get(1);
+            RelationshipType relationshipType = relationshipTypeService
+                .find(context, Integer.parseInt(req.getParameter("relationshipType")));
+
+            EPerson ePerson = context.getCurrentUser();
+            if (authorizeService.authorizeActionBoolean(context, leftItem, Constants.WRITE) ||
+                authorizeService.authorizeActionBoolean(context, rightItem, Constants.WRITE)) {
```

Comment:
These same authorization checks already occur in `relationshipService.create()`.  Should we just catch any `AuthorizeException` thrown and wrap it in an `AccessDeniedException`?

Reply:
This has been adjusted